packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -45,20 +46,36 @@ export function deprecatedBuildCustomElementConstructor(
     return Ctor.CustomElementConstructor;
 }
 
+// Note: WeakSet is not supported in IE11, and the polyfill is not performant enough.
+//       This WeakSet usage is valid because this functionality is not meant to run in IE11.
+const hydratedCustomElements = new WeakSet<Element>();
```

Comment:
well, what happen when LWC runs in IE11 then? this line will throw, isn't it?

Reply:
if you include the dist module, then yes. but is my understanding that LWC is not meant to run in IE11 without compat polyfills & downgrade; that's why the ie11 karma and integration tests are passing for this pr.